### DISCUSSION TITLE
Xiaomi IMILAB EC2

### DISCUSSION THREAD

**Discussion by tv4you2016** [roleoroleo/yi-hack-Allwinner-v2#590] [Q&A]
hello,
I've gone round and round and I can't put an RTSP server on this camera, at this moment I managed to backup all the partitions, I would like to know if someone can help me how to put an RTSP server
@roleoroleo  I know that with your abilities and with a little time you will be able to help me :D
[backup.zip](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10874977/backup.zip)

I also leave a small log that may help

[tmp_message.txt](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10874983/tmp_message.txt)

**Comment by roleoroleo** [comment]
Sorry but your cam is completely different from yi cams.
I can't help you with RTSP without a cam.

**Reply by tv4you2016** [reply]
Tell me what would you do if you were in my shoes where should I start?

**Reply by roleoroleo** [reply]
Study what the cam offers with default software.
I see some interesting libraries about stream and about mqtt.
Understand how the stream is sent to the user: does the cam use standard linux drivers (/dev/video0, video4linux, etc...) or something custom?
The 1st step is trying to understand how the cam works.

**Reply by tv4you2016** [reply]
root@hodor:~# ls -la /etc/init.d/
drwxr-xr-x    1 root     root             0 Nov  8 19:17 .
drwxr-xr-x    1 root     root             0 Mar  6  2020 ..
-rwxrwxr-x    1 root     root          1411 Sep 14  2018 boot
-rwxr-xr-x    1 root     root           407 Mar  6  2020 cm-ipcamera
-rwxr-xr-x    1 root     root           729 Mar  6  2020 cron
-rwxr-xr-x    1 root     root         14350 Mar  6  2020 dnsmasq
-rwxrwxr-x    1 root     root           265 Sep 14  2018 done
-rwxr-xr-x    1 root     root           997 Mar  6  2020 firewall
-rwxr-xr-x    1 root     root           341 Mar  6  2020 fstab
-rwxr-xr-x    1 root     root           144 Mar  6  2020 ipcamera-daemon
-rwxrwxr-x    1 root     root          2025 Sep 14  2018 led
-rwxr-xr-x    1 root     root          2169 Mar  6  2020 log
-rwxr-xr-x    1 root     root           918 Nov  8 23:30 mosquitto
-rwxr-xr-x    1 root     root          3253 Apr 11  2019 network
-rwxr-xr-x    1 root     root           244 Mar  6  2020 odhcpd
-rwxrwxr-x    1 root     root           125 Sep 14  2018 sysctl
-rwxrwxr-x    1 root     root           247 Sep 14  2018 sysfixtime
-rwxr-xr-x    1 root     root           770 Mar  6  2020 sysntpd
-rwxrwxr-x    1 root     root          1025 Sep 14  2018 system
-rwxr-xr-x    1 root     root           926 Mar  6  2020 telnet
-rwxr-xr-x    1 root     root           571 Mar  6  2020 test-hub
-rwxr-xr-x    1 root     root          3465 Mar  6  2020 uhttpd
-rwxrwxr-x    1 root     root           106 Sep 14  2018 umount

which one should i explore?

**Reply by tv4you2016** [reply]
root@hodor:~# netstat -tulpn | grep LISTEN
tcp        0      0 10.17.1.1:8000          0.0.0.0:*               LISTEN      1281/cm-ipcamera
tcp        0      0 10.17.1.1:80            0.0.0.0:*               LISTEN      1082/uhttpd
tcp        0      0 127.0.0.1:54322         0.0.0.0:*               LISTEN      1324/miio_client
tcp        0      0 127.0.0.1:54323         0.0.0.0:*               LISTEN      1324/miio_client
tcp        0      0 0.0.0.0:53              0.0.0.0:*               LISTEN      1306/dnsmasq
tcp        0      0 0.0.0.0:23              0.0.0.0:*               LISTEN      1050/telnetd
tcp        0      0 127.0.0.1:1883          0.0.0.0:*               LISTEN      952/mosquitto

**Reply by roleoroleo** [reply]
It seems a cam made by moobox: https://www.imoobox.com
Based on openwrt 14.07
There is a http server: check what you are able to do with it.

**Reply by tv4you2016** [reply]
I did second uhttpd instance on port 8881 for accessing the media files
and now?

**Reply by roleoroleo** [reply]
Is there a web interface with configuration options?

**Reply by tv4you2016** [reply]
no
